### ISSUE TITLE
Use PIN protection

### ISSUE THREAD

**Issue by aylen384** [psanford/tpm-fido#25]
I'm not sure if this is applicable here, but it would be nice, if it would optionally be possible to protect the tpm by a PIN that is requested from the user via pinentry when authenticating.

**Comment by psanford** [comment]
Yup, I plan on adding that for user-verification as part of supporting fido2.

**Comment by 497a** [comment]
I also would be happy about this feature! I like the idea of FDO but the idea of anyone just having to press 'OK' instead of having to provide a password bugs me a bit. I'd like to use passwordless login but no pin would mean unlimited access to my services for anyone with physical access to my device.

**Comment by PaarthShah** [comment]
Is there any particular set of things a potential contributor could look at to help support PINs?
Asking for a friend :^)

**Comment by guiand888** [comment]
Also looking forward to this feature! :)
Would be even nicer to integrate with PAM too as I assume we'd then be able to authenticate with a fingerprint reader.

**Comment by Tyg-g** [comment]
I also join the request for a PIN protection

**Comment by idkreally001** [comment]
This is implemented in a fork: https://github.com/idkreally001/tpm-fido — an optional PIN, requested via pinentry at authentication time (CTAP2 clientPIN). The PIN itself is never stored, only left-16-bytes-of-SHA-256(PIN) plus a retry counter, and it can be set/changed from a small tray app. There's also a "Windows Hello" mode where the PIN prompt is tpm-fido's own dialog rather than the browser's. See #48.
